src/Enzo/gravity/EnzoMethodBackgroundAcceleration.cpp

```diff
@@ -11,281 +11,230 @@
 
-  // Obtain grid sizes and ghost sizes
+  inline double z_val(int iz) const noexcept
+  { return lower_edge[2] + (iz - ghost_depth[2] + 0.5) * cell_width[2]; }
 
-  field.dimensions (0,&mx_,&my_,&mz_);
-  field.ghost_depth(0,&gx_,&gy_,&gz_);
-  double xp, yp, zp;
-  block->data()->lower(&xm_,&ym_,&zm_);
-  block->data()->upper(&xp,&yp,&zp);
-  field.cell_width(xm_,xp,&hx_,ym_,yp,&hy_,zm_,zp,&hz_);
+};
 
-  // We will probably never be in the situation of constant acceleration
-  // and cosmology, but just in case.....
-  EnzoPhysicsCosmology * cosmology = enzo::cosmology();
-  enzo_float cosmo_a = 1.0;
+//---------------------------------------------------------------------
 
-  const int rank = cello::rank();
+class GalaxyModel {
 
-  if (cosmology) {
-    enzo_float cosmo_dadt = 0.0;
-    double dt    = block->dt();
-    double time  = block->time();
-    cosmology->compute_expansion_factor(&cosmo_a,&cosmo_dadt,time+0.5*dt);
-    if (rank >= 1) hx_ *= cosmo_a;
-    if (rank >= 2) hy_ *= cosmo_a;
-    if (rank >= 3) hz_ *= cosmo_a;
-  }
+  /// @class    GalaxyModel
+  /// @ingroup  Enzo
+  /// @brief    [\ref Enzo] Evaluates the background potential of a Galaxy model
+  ///
+  /// If we didn't support cosmological simulations, we would simply store the
+  /// fields of the EnzoPotentialConfigGalaxy struct as members of this class
+  /// (in terms of code units)
 
+public:
+  GalaxyModel(const EnzoPotentialConfigGalaxy& pack_dfltU, const Units* units)
+  {
+    pack_codeU_ = EnzoPotentialConfigGalaxy::to_codeU(pack_dfltU, units);
 
-  enzo_float * ax = (enzo_float*) field.values ("acceleration_x");
-  enzo_float * ay = (enzo_float*) field.values ("acceleration_y");
-  enzo_float * az = (enzo_float*) field.values ("acceleration_z");
+    double xtemp = pack_codeU_.DM_mass_radius / pack_codeU_.rcore;
 
-  int m = mx_ * my_ * mz_;
-  if (zero_acceleration_){
-    if (ax){ for(int i = 0; i < m; i ++){ ax[i] = 0.0;}}
-    if (ay){ for(int i = 0; i < m; i ++){ ay[i] = 0.0;}}
-    if (az){ for(int i = 0; i < m; i ++){ az[i] = 0.0;}}
+    // compute the density constant for an NFW halo (rho_o)
+    DM_density_ = (pack_codeU_.DM_mass / (4.0 * cello::pi * std::pow(pack_codeU_.rcore,3))) / (std::log(1.0+xtemp)-xtemp/(1.0+xtemp));
   }
 
-  Particle particle = enzo_block->data()->particle();
-
-  if (enzo_config->method_background_acceleration_flavor == "GalaxyModel"){
-
-    this->GalaxyModel(ax, ay, az, &particle, rank,
-                      cosmo_a, enzo_config, enzo_units, enzo_block->dt);
-
-  } else if (enzo_config->method_background_acceleration_flavor == "PointMass"){
-    this->PointMass(ax, ay, az, &particle, rank,
-                    cosmo_a, enzo_config, enzo_units, enzo_block->dt);
-  } else {
-
-    ERROR("EnzoMethodBackgroundAcceleration::compute_()",
-          "Background acceleration flavor not recognized");
+  /// compute the x,y,z components of the acceleration
+  ///
+  /// @note
```

Comment:
Setting `accel_R` and `accel_z` to 0 for fluids was introduced by [Andrew Emerick back in 2019](https://github.com/enzo-project/enzo-e/commit/147d242a5d95352ed7877ed0843f586d3e2a6c40). According to the commit message, the intention was to create a more stable gas disk. I suspect he was attempting to dodge issues with immediately collapsing disk ICs, which is a problem I have struggled with for isolated galaxies in Enzo as well.

This is, in my opinion, a bad hack that could have terrible unforeseen consequences for anyone using Enzo-E that doesn't expect this to be present. For example, if I were to copy my isolated galaxy CGM sims over to Enzo-E, the gas of the entire CGM would also be affected by this hack and therefore wouldn't infall onto the disk correctly. If I had not reviewed this PR, this would be a challenging issue to debug.

If we want to keep this option for galaxy simulations, it should be parameterized. Otherwise I suggest it be removed and fluids and particles should be impacted by gravity equally.

Reply:
Fixed!

**Thanks for catching this and bringing this up!** I thought that this was a little funky way back when I first refactored the code. But, I didn't have much experience running simulations with gravity or simulations of galaxies (so I assumed that maybe I was missing something) and then I totally forgot about this.

Since then, I have done a lot of work on self-gravitating isolated galaxies in Cholla. In fact, I spent the entirety of this past February dealing with debugging gravitational instabilities in a self-gravitating gas disk (a little time was spent on vertical instability, but most of the time was related to radial instabilities).

**I wholeheartedly agree** that this hack is detrimental. I have removed the code entirely from the codebase.